### PR TITLE
Py3k

### PR DESCRIPTION
**Pull request by rowanc1**
Clone of #400 

Python 2 and Python 3 support for SimPEG!
-All old styled relative imports have been changed to either absolute imports or the more explicit styled relative imports
-All print statements have been changed to print function
-All instances of "except Exception, e:" changed to "except Exception as e:"
-Added version dependent type checking (Python 3 no longer has a "long" type)
-Caught issues of the new Python3 string representation when appending to an already opened file with np.savetxt(...)
-uses six package for cross-compatible metaclass implementation
-Anything else I can't remember

All python 2.7 tests pass on travis-ci

2 python3 tests fail:
-pip does not have vtk 7.0.0 which has python3 support, so vtk IO tests fail
-doc tests fail on Travis, not quite sure why as I can't replicate it on my system

P.S. the mump solvers wrapper package should probably also be updated to python3 cross-compatibility which should be easy.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -57,8 +57,7 @@ install:
       git clone https://github.com/rowanc1/pymatsolver.git;
       cd pymatsolver; python setup.py install; cd ..;
     else
-      git clone https://github.com/dwfmarchant/pyMKL.git;
-      cd pyMKL; python setup.py install; cd ..;
+      echo "pymatsolver not updated for python3";
```
Comment (rowanc1) [inline .travis.yml]: @jcapriot did you completely remove all the mkl/paradiso stuff from this PR?

Comment (jcapriot) [inline .travis.yml]: I reverted back to the previous so it should have undone everything.

**Comment by rowanc1** [comment]
image


The tests seem to be working on my computer. There does seem to be some issues however with the FDEM deriv tests? Does anyone have an idea of why this might not work?

**Comment by jcapriot** [comment]
Can we do a check in the pardiso solver interface to ensure that the sparse matrix is in a CSR format, and that it's "has_sorted_indices" property is true? I don't see that last check in the pyMKL package.

**Comment by rowanc1** [comment]
@bsmithyman do you have any off the cuff ideas?!

**Comment by rowanc1** [comment]
At least the fail is consistent now.

File: SimPEG/EM/Static/Utils/StaticUtils.py [hunk 2]
```diff
@@ -312,7 +313,7 @@ def xy_2_r(x1,x2,y1,y2):
             srcClass = DC.Src.Dipole([rxClass], M[0,:], N[-1,:])
         SrcList.append(srcClass)
     else:
-        print """stype must be either 'pole-dipole', 'dipole-dipole' or 'gradient'. """
+        print("""stype must be either 'pole-dipole', 'dipole-dipole' or 'gradient'. """)
```
Comment (lheagy) [inline SimPEG/EM/Static/Utils/StaticUtils.py]: Question: why 3 " ?

File: SimPEG/EM/Utils/testingUtils.py [hunk 3]
```diff
@@ -82,10 +83,11 @@ def getFDEMProblem(fdemType, comp, SrcList, freq, useMu=False, verbose=False):
     prb.pair(survey)
 
     try:
-        from pymatsolver import MumpsSolver
-        prb.Solver = MumpsSolver
-    except ImportError, e:
+        from pymatsolver import PardisoSolver
```
Comment (lheagy) [inline SimPEG/EM/Utils/testingUtils.py]: likely better to do the import at the top
```python
try: 
    from pymatsolver import PardisoSolver as Solver
except ImportError:
    from SimPEG import SolverLU as Solver
```

File: SimPEG/Examples/DC_Analytic_Dipole.py [hunk 4]
```diff
@@ -27,9 +30,9 @@ def run(plotIt=True):
     problem = DC.Problem3D_CC(mesh)
     problem.pair(survey)
     try:
-        from pymatsolver import MumpsSolver
-        problem.Solver = MumpsSolver
-    except Exception, e:
+        from pymatsolver import PardisoSolver
+        problem.Solver = PardisoSolver
```
Comment (lheagy) [inline SimPEG/Examples/DC_Analytic_Dipole.py]: same here, we should move imports to top of file

File: SimPEG/Mesh/MeshIO.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import numpy as np, os
 from SimPEG import Utils
+import six
```
Comment (lheagy) [inline SimPEG/Mesh/MeshIO.py]: has this been added to requirements.txt? (should it be?)

File: SimPEG/Mesh/View.py [hunk 6]
```diff
@@ -675,41 +679,6 @@ def plotImage(self, I, ax=None, showIt=False, grid=False, clim=None):
...
         ax.set_xlabel('x')
         ax.set_ylabel('y')
-        if showIt: plt.show()
+        if showIt:
+            plt.show()
         return [scalarMap]
-
-
-if __name__ == '__main__':
-    import numpy as np
```
Comment (lheagy) [inline SimPEG/Mesh/View.py]: did this get added back somewhere as an example? @rowanc1

File: SimPEG/EM/Static/Utils/StaticUtils.py (hunk 2 above)
Comment (rowanc1) [inline SimPEG/EM/Static/Utils/StaticUtils.py]: Could be fixed, might have been multiline before?

File: tests/em/fdem/forward/test_FDEM_analytics.py [hunk 7]
```diff
@@ -48,9 +49,9 @@ def setUp(self):
         prb.pair(survey)
 
         try:
-            from pymatsolver import MumpsSolver
-            prb.Solver = MumpsSolver
-        except ImportError, e:
+            from pymatsolver import PardisoSolver
```
Comment (lheagy) [inline tests/em/fdem/forward/test_FDEM_analytics.py]: same here - to top of file

**Comment by lheagy** [comment]
@jcapriot , @rowanc1 : do you have some recommendations for tutorials on python 3, relative imports etc? It would be great if we could include these in the [simpeg/tutorials](https://github.com/simpeg/tutorials) to support the development team!

File: SimPEG/Mesh/View.py (hunk 6 above)
Comment (rowanc1) [inline SimPEG/Mesh/View.py]: #403

**Pull request merged**